torch/csrc/jit/python/pybind_utils.h

```diff
@@ -869,6 +869,15 @@ inline IValue argumentToIValue(
             py::repr(object)),
         "\nCast error details: ",
         error.what()));
+  } catch (const py::error_already_set& error) {
```

Comment:
when will we hit this new path?

Reply:
In operator matching from RPC, we can get errors such as `TypeError: 'int' object is not iterable` when we try to match operators that are not a match. We were previously swallowing these exceptions and handling them with VLOG in the RPC code, but they are errors during schema match, so we convert them appropriately to schema match errors.

Ideally, this would surface as type error but `catch (const py::type_error)` does not catch the error, but py::error_already_set does catch it.